### PR TITLE
repair broken consensus change distribution

### PR DESCRIPTION
**Pull request by DavidVorick**
By combining consensus changes before distributing them when downloading multiple blocks at a time, we ended up providing subscribers with consensus change IDs that did not match the original consensus change ID. The result was a lot of the modules started resetting themselves.

This fixes that, I should have thought about it on the first pass, and I remember being skeptical that it was the right way to go.

Code has been simplified, and now it should be correct.

We need to replace RC3 with this code ASAP, it's a pretty massive usability fix.

### CODE REVIEW

**Comment by DavidVorick** [comment]
Actually, let's keep this open until I have a regression test for it.

**Comment by DavidVorick** [comment]
Should be good to go, including regtests.

File: modules/consensus/accept_test.go [hunk 1]
```diff
@@ -1145,7 +1148,22 @@ func TestChainedAcceptBlock(t *testing.T) {
...
-	if bcs.changes != 3 || bcs.appliedBlocks != int(cst2.cs.Height()+1) || bcs.revertedBlocks != 0 {
+	if bcs.appliedBlocks != int(cst2.cs.Height()+1) || bcs.revertedBlocks != 0 {
 		t.Error("consensus changes do not seem to be getting passed to subscribers correctly")
 	}
+
+	// Check that every change recorded in 'bcs' is also available in the
+	// consensus set.
+	for _, change := range bcs.changes {
+		err := cst2.cs.db.Update(func(tx *bolt.Tx) error {
+			_, exists := getEntry(tx, change)
```
Comment (lukechampine) [inline modules/consensus/accept_test.go]: shouldn't this be a `View`? And shouldn't the `for` loop be inside it?

Comment (lukechampine) [inline modules/consensus/accept_test.go]: oh, it's in a test so not a big deal

Comment (DavidVorick) [inline modules/consensus/accept_test.go]: ah yeah, little lazy about the performance here. Maybe adds 1 second to the test, shouldn't be too bad.

**Pull request merged**